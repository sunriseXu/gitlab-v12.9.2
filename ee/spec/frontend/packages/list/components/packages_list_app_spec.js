import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { GlEmptyState, GlTab, GlTabs } from '@gitlab/ui';
import PackageListApp from 'ee/packages/list/components/packages_list_app.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('packages_list_app', () => {
  let wrapper;
  let store;

  const PackageList = {
    name: 'package-list',
    template: '<div><slot name="empty-state"></slot></div>',
  };
  const GlLoadingIcon = { name: 'gl-loading-icon', template: '<div>loading</div>' };

  const emptyListHelpUrl = 'helpUrl';
  const findListComponent = () => wrapper.find(PackageList);
  const findTabComponent = (index = 0) => wrapper.findAll(GlTab).at(index);

  const mountComponent = () => {
    wrapper = shallowMount(PackageListApp, {
      localVue,
      store,
      stubs: {
        GlEmptyState,
        GlLoadingIcon,
        PackageList,
        GlTab,
        GlTabs,
      },
    });
  };

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isLoading: false,
        config: {
          resourceId: 'project_id',
          emptyListIllustration: 'helpSvg',
          emptyListHelpUrl,
        },
      },
    });
    store.dispatch = jest.fn();
    mountComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders', () => {
    mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('empty state', () => {
    it('generate the correct empty list link', () => {
      const link = findListComponent().find('a');

      expect(link.element).toMatchInlineSnapshot(`
        <a
          href="${emptyListHelpUrl}"
          target="_blank"
        >
          publish and share your packages
        </a>
      `);
    });

    it('includes the right content on the default tab', () => {
      const heading = findListComponent().find('h4');

      expect(heading.text()).toBe('There are no packages yet');
    });
  });

  it('call requestPackagesList on page:changed', () => {
    const list = findListComponent();
    list.vm.$emit('page:changed', 1);
    expect(store.dispatch).toHaveBeenCalledWith('requestPackagesList', { page: 1 });
  });

  it('call requestDeletePackage on package:delete', () => {
    const list = findListComponent();
    list.vm.$emit('package:delete', 'foo');
    expect(store.dispatch).toHaveBeenCalledWith('requestDeletePackage', 'foo');
  });

  it('calls requestPackagesList on sort:changed', () => {
    const list = findListComponent();
    list.vm.$emit('sort:changed');
    expect(store.dispatch).toHaveBeenCalledWith('requestPackagesList');
  });

  describe('tab change', () => {
    it('calls requestPackagesList when all tab is clicked', () => {
      findTabComponent().trigger('click');

      expect(store.dispatch).toHaveBeenCalledWith('requestPackagesList');
    });

    it('calls requestPackagesList when a package type tab is clicked', () => {
      findTabComponent(1).trigger('click');

      expect(store.dispatch).toHaveBeenCalledWith('requestPackagesList');
    });
  });
});
