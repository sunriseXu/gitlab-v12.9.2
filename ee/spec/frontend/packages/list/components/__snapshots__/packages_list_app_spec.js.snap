// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packages_list_app renders 1`] = `
<b-tabs-stub
  activenavitemclass="gl-tab-nav-item-active gl-tab-nav-item-active-indigo"
  class="gl-tabs"
  contentclass=",gl-tab-content"
  navclass="gl-tabs-nav"
  nofade="true"
  nonavstyle="true"
  tag="div"
>
  <template>
     
    <b-tab-stub
      buttonid=""
      tag="div"
      title="All"
      titlelinkclass="gl-tab-nav-item"
    >
      <template>
        <div>
          <div
            class="row empty-state"
          >
            <div
              class="col-12"
            >
              <div
                class="svg-250 svg-content"
              >
                <img
                  alt="There are no packages yet"
                  class=""
                  src="helpSvg"
                />
              </div>
            </div>
             
            <div
              class="col-12"
            >
              <div
                class="text-content"
              >
                <h4
                  class="center"
                  style=""
                >
                  There are no packages yet
                </h4>
                 
                <p
                  class="center"
                  style=""
                >
                  <p>
                    Learn how to 
                    <a
                      href="helpUrl"
                      target="_blank"
                    >
                      publish and share your packages
                    </a>
                     with GitLab.
                  </p>
                </p>
                 
                <div
                  class="text-center"
                >
                  <!---->
                   
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </template>
    </b-tab-stub>
    <b-tab-stub
      buttonid=""
      tag="div"
      title="Conan"
      titlelinkclass="gl-tab-nav-item"
    >
      <template>
        <div>
          <div
            class="row empty-state"
          >
            <div
              class="col-12"
            >
              <div
                class="svg-250 svg-content"
              >
                <img
                  alt="There are no Conan packages yet"
                  class=""
                  src="helpSvg"
                />
              </div>
            </div>
             
            <div
              class="col-12"
            >
              <div
                class="text-content"
              >
                <h4
                  class="center"
                  style=""
                >
                  There are no Conan packages yet
                </h4>
                 
                <p
                  class="center"
                  style=""
                >
                  <p>
                    Learn how to 
                    <a
                      href="helpUrl"
                      target="_blank"
                    >
                      publish and share your packages
                    </a>
                     with GitLab.
                  </p>
                </p>
                 
                <div
                  class="text-center"
                >
                  <!---->
                   
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </template>
    </b-tab-stub>
    <b-tab-stub
      buttonid=""
      tag="div"
      title="Maven"
      titlelinkclass="gl-tab-nav-item"
    >
      <template>
        <div>
          <div
            class="row empty-state"
          >
            <div
              class="col-12"
            >
              <div
                class="svg-250 svg-content"
              >
                <img
                  alt="There are no Maven packages yet"
                  class=""
                  src="helpSvg"
                />
              </div>
            </div>
             
            <div
              class="col-12"
            >
              <div
                class="text-content"
              >
                <h4
                  class="center"
                  style=""
                >
                  There are no Maven packages yet
                </h4>
                 
                <p
                  class="center"
                  style=""
                >
                  <p>
                    Learn how to 
                    <a
                      href="helpUrl"
                      target="_blank"
                    >
                      publish and share your packages
                    </a>
                     with GitLab.
                  </p>
                </p>
                 
                <div
                  class="text-center"
                >
                  <!---->
                   
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </template>
    </b-tab-stub>
    <b-tab-stub
      buttonid=""
      tag="div"
      title="NPM"
      titlelinkclass="gl-tab-nav-item"
    >
      <template>
        <div>
          <div
            class="row empty-state"
          >
            <div
              class="col-12"
            >
              <div
                class="svg-250 svg-content"
              >
                <img
                  alt="There are no NPM packages yet"
                  class=""
                  src="helpSvg"
                />
              </div>
            </div>
             
            <div
              class="col-12"
            >
              <div
                class="text-content"
              >
                <h4
                  class="center"
                  style=""
                >
                  There are no NPM packages yet
                </h4>
                 
                <p
                  class="center"
                  style=""
                >
                  <p>
                    Learn how to 
                    <a
                      href="helpUrl"
                      target="_blank"
                    >
                      publish and share your packages
                    </a>
                     with GitLab.
                  </p>
                </p>
                 
                <div
                  class="text-center"
                >
                  <!---->
                   
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </template>
    </b-tab-stub>
    <b-tab-stub
      buttonid=""
      tag="div"
      title="NuGet"
      titlelinkclass="gl-tab-nav-item"
    >
      <template>
        <div>
          <div
            class="row empty-state"
          >
            <div
              class="col-12"
            >
              <div
                class="svg-250 svg-content"
              >
                <img
                  alt="There are no NuGet packages yet"
                  class=""
                  src="helpSvg"
                />
              </div>
            </div>
             
            <div
              class="col-12"
            >
              <div
                class="text-content"
              >
                <h4
                  class="center"
                  style=""
                >
                  There are no NuGet packages yet
                </h4>
                 
                <p
                  class="center"
                  style=""
                >
                  <p>
                    Learn how to 
                    <a
                      href="helpUrl"
                      target="_blank"
                    >
                      publish and share your packages
                    </a>
                     with GitLab.
                  </p>
                </p>
                 
                <div
                  class="text-center"
                >
                  <!---->
                   
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </template>
    </b-tab-stub>
  </template>
  <template>
    <div
      class="align-self-center ml-auto"
    >
      <package-sort-stub />
    </div>
  </template>
</b-tabs-stub>
`;
